services/agent/src/simcore_service_agent/services/containers_manager.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from dataclasses import dataclass, field
+
+from aiodocker import Docker
+from fastapi import FastAPI
+from models_library.api_schemas_directorv2.services import (
+    DYNAMIC_PROXY_SERVICE_PREFIX,
+    DYNAMIC_SIDECAR_SERVICE_PREFIX,
+)
+from models_library.projects_nodes_io import NodeID
+from servicelib.fastapi.app_state import SingletonInAppStateMixin
+from servicelib.utils import limited_gather
+
+from ..core.settings import ApplicationSettings
+from .docker_utils import get_containers_with_prefixes, remove_container_forcefully
+
+_logger = logging.getLogger(__name__)
+_always_visible_logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ContainersManager(SingletonInAppStateMixin):
+    app_state_name: str = "containers_manager"
+
+    docker: Docker = field(default_factory=Docker)
+
+    async def force_container_cleanup(self, node_id: NodeID) -> None:
+        # compose all possible used container prefixes
+        proxy_prefix = f"{DYNAMIC_PROXY_SERVICE_PREFIX}_{node_id}"
+        dy_sidecar_prefix = f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}_{node_id}"
+        user_service_prefix = f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}-{node_id}"
+
+        orphan_containers = await get_containers_with_prefixes(
```

Comment:
There are also permanent nodes in the systems, e.g. GPU1 on AWS-NIH-Prod. If two people both (like Taylor and his student) run services there, would'nt this assume both to be orphaned as the function is not aware of for example the UUID?

Reply:
Or, let me ask differently: Where is the service UUID passed?

Reply:
The prefix strings already contain the service uuid. If you look at the lines above(29 -> 31) they contain the `node_id` which is what you call service UUID.